src/jemalloc.c

```diff
@@ -1352,250 +1352,466 @@ malloc_init_hard(void)
  */
 /******************************************************************************/
 /*
- * Begin malloc(3)-compatible functions.
+ * Begin allocation-path internal functions and data structures.
  */
 
-static void *
-ialloc_prof_sample(tsd_t *tsd, size_t usize, szind_t ind, bool zero,
-    prof_tctx_t *tctx, bool slow_path)
-{
-	void *p;
+/* Settings determined by the documented behavior of the allocation functions */
+typedef struct static_opts_s static_opts_t;
```

Comment:
Feel free to combine this into ```typedef struct {...} static_opts_t``` (or leave it; your choice).  The headers are consistent about using separate declarations just so that forward declarations are easy, but here we'll never have that problem.